frameworks/cassandra/tests/test_jmx.py

```diff
@@ -0,0 +1,261 @@
+    create_keystore_cmd = [
+        "keytool",
+        "-genkey",
+        "-alias",
+        "self-signed-cert",
+        "-keyalg",
+        "rsa",
+        "-dname",
+        "CN=myhost.example.com,O=Example Company,C=US",
+        "-keystore",
+        "/tmp/{}-self-signed-keystore.ks".format(test_run),
+        "-storepass",
+        "deleteme",
+        "-keypass",
+        "deleteme",
+        "-storetype",
+        "jks",
+    ]
+
+    subprocess.check_output(create_keystore_cmd)
+
+    keystore_list_cmd = [
+        "keytool",
+        "-list",
+        "-v",
+        "-keystore",
+        "/tmp/{}-self-signed-keystore.ks".format(test_run),
+        "-storepass",
+        "deleteme",
+    ]
+
+    subprocess.check_output(keystore_list_cmd)
+
+    write_to_file("deleteme", "/tmp/{}-keystorepass".format(test_run))
+    write_to_file("admin adminpassword", "/tmp/{}-passwordfile".format(test_run))
+    write_to_file("admin readwrite", "/tmp/{}-access".format(test_run))
+
+    sdk_security.install_enterprise_cli(False)
+
+    sdk_cmd.run_cli(
+        "security secrets create -f /tmp/{}-self-signed-keystore.ks {}".format(test_run, KEY_STORE)
+    )
+    sdk_cmd.run_cli(
+        "security secrets create -f /tmp/{}-passwordfile {}".format(test_run, PASSWORD_FILE)
+    )
+    sdk_cmd.run_cli(
+        "security secrets create -f /tmp/{}-keystorepass {}".format(test_run, KEY_STORE_PASS)
+    )
+    sdk_cmd.run_cli("security secrets create -f /tmp/{}-access {}".format(test_run, ACCESS_FILE))
+
+
+def uninstall_jmx_secrets():
+    sdk_security.delete_secret(KEY_STORE)
+    sdk_security.delete_secret(KEY_STORE_PASS)
+    sdk_security.delete_secret(ACCESS_FILE)
+    sdk_security.delete_secret(PASSWORD_FILE)
+
+
+@pytest.mark.sanity
+@sdk_utils.dcos_ee_only
+@pytest.mark.parametrize("self_signed_trust_store", [True, False])
+@pytest.mark.parametrize("authentication", [True, False])
+def test_secure_jmx_configuration(self_signed_trust_store, authentication):
+    foldered_name = config.get_foldered_service_name()
+    test_jobs: List[Dict[str, Any]] = []
+    try:
+        test_jobs = config.get_all_jobs(node_address=config.get_foldered_node_address())
+        # destroy/reinstall any prior leftover jobs, so that they don't touch the newly installed service:
+        for job in test_jobs:
+            sdk_jobs.install_job(job)
+
+        install_jmx_configured_cassandra(
+            self_signed_trust_store=self_signed_trust_store, authentication=authentication
+        )
+        node_task_id_0 = sdk_tasks.get_task_ids(foldered_name)[0]
+        install_jmxterm(task_id=node_task_id_0)
+        generate_jmx_command_files(task_id=node_task_id_0)
+
+        if self_signed_trust_store:
+            trust_store = "$MESOS_SANDBOX/jmx/trust_store"
+            trust_store_password = "deleteme"
+        else:
+            trust_store = "$JAVA_HOME/lib/security/cacerts"
+            trust_store_password = "changeit"
+
+        if not authentication:
```

Comment:
How comes these tests only work when authentication is off?

Reply:
because test_backup_and_restore_to_s3  does not work with auth, as for auth we have separate test which have different dependencies. 
With auth we just need to check JMX is working or not, as in task ```init_system_keyspaces``` under auth  some changes were done as part of JMX. 
and with auth back/restore - repair/cleanup test are not required as JMX changes are already evaluating in non-auth back/restore - repair/cleanup.